app/models/project_step.rb

```diff
@@ -135,6 +138,27 @@ def handle_original_date_logic
     end
   end
 
+  def handle_finalized_at
```

Comment:
These methods should be private (as generally should any that don't need to be called from outside the class).